src/main/java/in/projecteka/gateway/common/DefaultValidatedResponseAction.java

```diff
@@ -11,26 +13,31 @@
 
 import java.util.Optional;
 
+import static in.projecteka.gateway.common.Constants.X_HIU_ID;
+
 @AllArgsConstructor
 public class DefaultValidatedResponseAction<T extends ServiceClient> implements ValidatedResponseAction{
     private static final Logger logger = LoggerFactory.getLogger(DefaultValidatedResponseAction.class);
     T serviceClient;
     CMRegistry cmRegistry;
+    BridgeRegistry bridgeRegistry;
 
     @Override
-    public Mono<Void> routeResponse(String xCmId, JsonNode updatedRequest) {
-        Optional<YamlRegistryMapping> configFor = cmRegistry.getConfigFor(xCmId);
+    public Mono<Void> routeResponse(String id, JsonNode updatedRequest) {
+        Optional<YamlRegistryMapping> configFor = id.equals(X_HIU_ID)
+                ? bridgeRegistry.getConfigFor(id, ServiceType.HIU)
+                : cmRegistry.getConfigFor(id);
         if (configFor.isEmpty()) {
-            logger.error("No mapping found for {}",xCmId);
+            logger.error("No mapping found for {}",id);
             return Mono.empty();
         }
         return serviceClient.routeResponse(updatedRequest,configFor.get().getHost());
     }
 
     @Override
-    public Mono<Void> handleError(Throwable throwable, String xCmId, JsonNode jsonNode) {
+    public Mono<Void> handleError(Throwable throwable, String id, JsonNode jsonNode) {
         //Does it make sense to call the same API back to notify only Error?
-        logger.error("Error in notifying CM with result",throwable);
+        logger.error("Error in notifying host with result",throwable);
         return Mono.empty();
```

Comment:
jsonNode is at all used. I see that it's coming from base class, if it's not used anywhere, just remove it please.

Reply:
It's been used at another place to push the jsonNode as a message to an exchange.